### DISCUSSION TITLE
Is it possible to blacklist all the .txt files recursively under a directory?

### DISCUSSION THREAD

**Discussion by YorkZ** [netblue30/firejail#5616] [Q&A]
In Bash filename expansion, if `globstar` is on by:

``` shell
shopt -s globstar
```

`**` can be used to match all the files recursively.

Would it be possible to blacklist all the `.txt` files recursively similarly, by something like:

    blacklist /path/to/foo/**/*.txt

?

**Comment by ghost** [comment]
Firejail does implement some file globbing. But it uses `glibc`'s glob command and AFAIK that doesn't support `**`. Would be a nice feature though.

I have used [ignorefs](https://github.com/ryanpetris/ignorefs) recently to hide specific files. You could prepare your filesystem with that before creating a firejail sandbox.

**Comment by kmk3** [comment]
@YorkZ [on Jan 25](https://github.com/netblue30/firejail/discussions/5616#discussion-4789840):

> Would it be possible to blacklist all the `.txt` files recursively similarly, by something like:
> 
> ```
> blacklist /path/to/foo/**/*.txt
> ```

No, but it's possible to generate a profile that does that and include it.
Example:

~/bin/someprogram (wrapper for someprogram):

```sh
#!/bin/sh
find /path/to/foo -type f -name '*.txt' | LC_ALL=C sort -u |
  sed 's/^/blacklist /' >~/.config/firejail/disable-txt.inc.local

firejail /usr/bin/someprogram
```

~/.config/firejail/someprogram.local:

```
include disable-txt.inc.local
```

Note: This cannot blacklist any files (.txt or otherwise) that may be added to
/path/to/foo after the program was started, so ideally whitelisting would be
used instead.

Note2: The `find` pipeline assumes that none of the files matched contain any
special characters that may cause issues.
